drivers/gpio/gpio_stm32.c

```diff
@@ -326,12 +336,21 @@ static void gpio_stm32_set_exti_source(int port, int pin)
 
 #elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32g0_exti)
 	LL_EXTI_SetEXTISource(port, line);
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7rs_exti)
+	/* 'port' is of LL_SBS_EXTI_PORTx, 'line' is the pin value [0..15] */
+	/* Using HAL formula when setting the gpio port during HAL_EXTI_SetConfigLine */
+	uint32_t regval = SBS->EXTICR[line >> 2U];
+
+	regval &= ~(SBS_EXTICR1_PC_EXTI0 << (SBS_EXTICR1_PC_EXTI1_Pos * (line & 0x03u)));
+	regval |= (port << (SBS_EXTICR1_PC_EXTI1_Pos * (line & 0x03u)));
+	SBS->EXTICR[line >> 2u] = regval;
```

Comment:
Why not using `LL_SBS_SetEXTISource` ?

Reply:
change formula to use LL_SBS_SetEXTISource